web-bundle/src/main/java/com/graphhopper/replica/ReplicaCustomSpeedsCarTagParser.java

```diff
@@ -0,0 +1,37 @@
+package com.graphhopper.replica;
+
+import com.google.common.collect.ImmutableMap;
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.routing.ev.EncodedValueLookup;
+import com.graphhopper.routing.util.CarTagParser;
+import com.graphhopper.util.PMap;
+
+import java.util.Objects;
+
+public class ReplicaCustomSpeedsCarTagParser extends CarTagParser {
+    private final ImmutableMap<Long, Double> osmWayIdToMaxSpeed;
+
+    public ReplicaCustomSpeedsCarTagParser(EncodedValueLookup lookup, PMap configuration, ImmutableMap<Long, Double> osmWayIdToMaxSpeed) {
+        super(lookup, configuration);
+        this.osmWayIdToMaxSpeed = osmWayIdToMaxSpeed;
+    }
+
+    @Override
+    protected double applyMaxSpeed(ReaderWay way, double speed) {
+        // n.b. superclass logic sets max speed to be 90% of the OSM's max speed for the way, but we don't apply the 90%
```

Comment:
Great comments here + in `applyBadSurfaceSpeed` - kind of an obvious note, but will be good to ensure these details are documented in whatever usage docs we end up creating for this feature (as well as the interaction between custom profiles + custom speeds that we discussed this morning)

Reply:
good call, noted those as things to mention in the upcoming usage doc!